wgpu-types/src/lib.rs

```diff
@@ -6383,10 +6383,19 @@ impl<L, V> TextureDescriptor<L, V> {
     ///
     /// <https://gpuweb.github.io/gpuweb/#abstract-opdef-compute-render-extent>
     #[must_use]
-    pub fn compute_render_extent(&self, mip_level: u32) -> Extent3d {
+    pub fn compute_render_extent(&self, mip_level: u32, plane: Option<u32>) -> Extent3d {
+        let width = self.size.width >> mip_level;
+        let height = self.size.height >> mip_level;
+
+        let (width, height) = match (self.format, plane) {
+            (TextureFormat::NV12 | TextureFormat::P010, Some(0)) => (width, height),
+            (TextureFormat::NV12 | TextureFormat::P010, Some(1)) => (width / 2, height / 2),
+            _ => (width, height),
+        };
```

Comment:
My only worry here is that when we add future multi-planar formats, there will be no error to tell us to populate this. Maybe a (debug) assert in the catchall branch to assert that the format is single-planar and maybe a test which calls this function on all texture formats?

Reply:
https://github.com/gfx-rs/wgpu/issues/8438